File: react-native/src/network/GuestApiClient.ts
/**
 * GuestApiClient.ts
 * Standalone fetch client for guest user initialization.
 * Operates without auth tokens — uses API-Key header only.
 */

import { API_INITIALIZE_USER, GUEST_API_KEY } from '../config/constants';

export interface InitializeGuestUserResponse {
  access_token: string;
  refresh_token: string;
  user_id?: string;
  created_now?: boolean;  // true = new user created, false = existing user returned for this device_id
  country_code?: string;
  country?: string;
  state?: string;
}

/**
 * POST api/user/initialize_user/
 * Returns access_token, refresh_token and user_id for a new guest device.
 * Throws on network error or non-2xx response.
 */
export async function initializeGuestUser(
  baseUrl: string,
  deviceId: string,
): Promise<InitializeGuestUserResponse> {
  const url = `${baseUrl.replace(/\/$/, '')}/${API_INITIALIZE_USER}`;

  const response = await fetch(url, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
      Authorization: `Api-Key ${GUEST_API_KEY}`,
    },
    body: JSON.stringify({ device_id: deviceId }),
  });

  if (!response.ok) {
    const body = await response.text().catch(() => '');
    throw new Error(`initialize_user failed: HTTP ${response.status} — ${body}`);
  }

  return response.json() as Promise<InitializeGuestUserResponse>;
}

File: ios/Sources/FarmerChatSDK/Network/GuestAPIClient.swift
import Foundation

// MARK: - GuestAPIClient

/// Standalone URLSession client for guest user initialization.
/// Operates independently of the main authenticated API client — no tokens required.
internal enum GuestAPIClient {

    private static let apiKey = "Y2K3kW5R9uQ0fL2X8zI7hT3aJ7"

    /// POST api/user/initialize_user/
    /// Obtains access_token, refresh_token, and user_id for a device on first install.
    static func initializeUser(
        baseUrl: String,
        deviceId: String
    ) async throws -> InitializeGuestUserResponse {
        let base = baseUrl.trimmingCharacters(in: CharacterSet(charactersIn: "/"))
        let urlString = base + "/" + APIEndpoints.initializeUser

        guard let url = URL(string: urlString) else {
            throw URLError(.badURL)
        }

        var request = URLRequest(url: url)
        request.httpMethod = "POST"
        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
        request.setValue("Api-Key \(apiKey)", forHTTPHeaderField: "Authorization")
        request.timeoutInterval = 30

        let encoder = JSONEncoder()
        request.httpBody = try encoder.encode(InitializeGuestUserRequest(device_id: deviceId))

        let (data, response) = try await URLSession.shared.data(for: request)

        guard let httpResponse = response as? HTTPURLResponse else {
            throw URLError(.badServerResponse)
        }
        guard httpResponse.statusCode == 200 else {
            throw URLError(.badServerResponse)
        }

        return try JSONDecoder().decode(InitializeGuestUserResponse.self, from: data)
    }
}
